Sources/_SwiftSyntaxTestSupport/FixItApplier.swift

```diff
@@ -0,0 +1,149 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftDiagnostics
+import SwiftSyntax
+
+public enum FixItApplier {
+  fileprivate struct Edit: Equatable {
+    var startUtf8Offset: Int
+    var endUtf8Offset: Int
+    let replacement: String
+
+    var originalLength: Int {
+      return endUtf8Offset - startUtf8Offset
+    }
+
+    var replacementLength: Int {
+      return replacement.utf8.count
+    }
+
+    var replacementRange: Range<Int> {
+      return startUtf8Offset..<endUtf8Offset
+    }
+  }
+
+  /// Applies selected or all Fix-Its from the provided diagnostics to a given syntax tree.
+  ///
+  /// - Parameters:
+  ///   - diagnostics: An array of `Diagnostic` objects, each containing one or more Fix-Its.
+  ///   - filterByMessages: An optional array of message strings to filter which Fix-Its to apply.
+  ///                      If `nil`, all Fix-Its in `diagnostics` are applied.
+  ///   - tree: The syntax tree to which the Fix-Its will be applied.
+  ///
+  /// - Returns: A ``String`` representation of the modified syntax tree after applying the Fix-Its.
+  public static func applyFixes(
+    from diagnostics: [Diagnostic],
+    filterByMessages messages: [String]?,
+    to tree: any SyntaxProtocol
+  ) -> String {
+    let messages = messages ?? diagnostics.compactMap { $0.fixIts.first?.message.message }
+
+    let changes =
+      diagnostics
+      .flatMap(\.fixIts)
+      .filter { messages.contains($0.message.message) }
+      .flatMap(\.changes)
+
+    var edits: [Edit] = []
+
+    for change in changes {
+      switch change {
+      case .replace(let oldNode, let newNode):
+        edits.append(
+          Edit(
+            startUtf8Offset: oldNode.position.utf8Offset,
+            endUtf8Offset: oldNode.endPosition.utf8Offset,
+            replacement: newNode.description
+          )
+        )
+
+      case .replaceLeadingTrivia(let token, let newTrivia):
+        edits.append(
+          Edit(
+            startUtf8Offset: token.position.utf8Offset,
+            endUtf8Offset: token.positionAfterSkippingLeadingTrivia.utf8Offset,
+            replacement: newTrivia.description
+          )
+        )
+
+      case .replaceTrailingTrivia(let token, let newTrivia):
+        edits.append(
+          Edit(
+            startUtf8Offset: token.endPositionBeforeTrailingTrivia.utf8Offset,
+            endUtf8Offset: token.endPosition.utf8Offset,
+            replacement: newTrivia.description
+          )
+        )
+      }
```

Comment:
What do you think about making this `switch` an extension on `Change`?

I could very selfishly use that for https://github.com/apple/sourcekit-lsp/pull/902/files#diff-6147b806788b196876d46885643488f44c48b14791d7adc3ef5d2d91b361bd11R74-R76 😉